tests/test_helpers.py

```diff
@@ -0,0 +1,133 @@
+
+
+class NodeTreeSearchTestCase(CMSTestCase):
+
+    def setUp(self):
+        self.language = settings.LANGUAGES[0][0]
+
+    def test_page_content_type_in_node_tree(self):
+        """
+        Test to ensure correct node mapped to page content object is returned from node tree search
+        """
+        menu_contents = factories.MenuContentFactory()
+        child1 = factories.ChildMenuItemFactory(parent=menu_contents.root)
+        page_content = factories.PageContentWithVersionFactory(
+            language=self.language, version__created_by=self.get_superuser()
+        )
+        child2 = factories.ChildMenuItemFactory(parent=menu_contents.root, content=page_content)
+
+        result = search_content_object_from_node_tree(menu_contents, page_content)
+
+        self.assertEqual(result, child2)
+
+    def test_to_search_content_object_in_nested_node_tree_search(self):
+        """
+        Test to ensure first node is returned  if more than one node is mapped to content object from node tree search
+        """
+        menu_contents = factories.MenuContentFactory()
+        child1 = factories.ChildMenuItemFactory(parent=menu_contents.root)
+        page_content = factories.PageContentWithVersionFactory(
+            language=self.language, version__created_by=self.get_superuser()
+        )
+        child2 = factories.ChildMenuItemFactory(parent=menu_contents.root)
+        grandchild = factories.ChildMenuItemFactory(parent=child1)
+        grandchild1 = factories.ChildMenuItemFactory(parent=grandchild)
+        grandchild2 = factories.ChildMenuItemFactory(parent=grandchild1, content=page_content)
+
+        result = search_content_object_from_node_tree(menu_contents, page_content)
+
+        self.assertEqual(result, grandchild2)
+
+    def test_content_object_not_found_in_node_tree(self):
+        """
+        Test to ensure False if returned  when no node is mapped to content object from node tree search
+        """
+        menu_contents = factories.MenuContentFactory()
+        child1 = factories.ChildMenuItemFactory(parent=menu_contents.root)
+        page_content = factories.PageContentWithVersionFactory(
+            language=self.language, version__created_by=self.get_superuser()
+        )
+        child2 = factories.ChildMenuItemFactory(parent=menu_contents.root)
+
+        result = search_content_object_from_node_tree(menu_contents, page_content)
+
+        self.assertFalse(result)
+
+        grandchild = factories.ChildMenuItemFactory(parent=child1)
+        grandchild1 = factories.ChildMenuItemFactory(parent=grandchild)
+        grandchild2 = factories.ChildMenuItemFactory(parent=grandchild1)
+
+        result = search_content_object_from_node_tree(menu_contents, page_content)
+
+        self.assertFalse(result)
+
+    def test_content_object_mapped_to_more_than_node(self):
+        """
+        Test to ensure first node found returned when more than one node is mapped to content object from node tree search
```

Comment:
Change the description to something like: "The first node found returned when more than one node is mapped to content object"